custom/icds_reports/utils/aggregation_helpers/distributed/agg_service_delivery_report.py

```diff
@@ -0,0 +1,256 @@
+                SUM(CASE WHEN thr_eligible=1 AND num_rations_distributed BETWEEN 1 AND 7 THEN 1 ELSE 0 END) as mother_thr_1_7_days,
+                SUM(CASE WHEN thr_eligible=1 AND num_rations_distributed BETWEEN 8 AND 14 THEN 1 ELSE 0 END) as mother_thr_8_14_days,
+                SUM(CASE WHEN thr_eligible=1 AND num_rations_distributed BETWEEN 15 AND 20 THEN 1 ELSE 0 END) as mother_thr_15_20_days,
+                SUM(CASE WHEN thr_eligible=1 AND num_rations_distributed>=21 THEN 1 ELSE 0 END) as mother_thr_21_days
+            FROM ccs_record_monthly
+            WHERE month=%(start_date)s
+            GROUP BY supervisor_id, awc_id
+        """,{
+            "start_date":self.month
+        }
+
+        yield """
+        UPDATE "{tmp_tablename}" agg_sdr
+        SET thr_eligible = thr_eligible +  ut.mother_thr_eligible,
+            thr_0_days = thr_0_days +  ut.mother_thr_0_days,
+            thr_1_7_days = thr_1_7_days +  ut.mother_thr_1_7_days,
+            thr_8_14_days = thr_8_14_days +  ut.mother_thr_8_14_days,
+            thr_15_20_days = thr_15_20_days +  ut.mother_thr_15_20_days,
+            thr_21_days = thr_21_days +  ut.mother_thr_21_days
+        FROM temp_ccs_table ut
+        WHERE agg_sdr.awc_id = ut.awc_id;
+        """.format(
+            tmp_tablename=self.staging_tablename,
+        ), {}
+
+        yield """
+        DROP TABLE temp_ccs_table;
+        """, {}
+
+    def rollup_query(self, aggregation_level):
+        columns = (
+            ('state_id', 'state_id'),
+            ('district_id', lambda col: col if aggregation_level > 1 else "'All'"),
+            ('block_id', lambda col: col if aggregation_level > 2 else "'All'"),
+            ('supervisor_id', lambda col: col if aggregation_level > 3 else "'All'"),
+            ('awc_id', lambda col: col if aggregation_level > 4 else "'All'"),
+            ('aggregation_level', str(aggregation_level)),
+            ('month', 'month'),
+            ('pse_eligible',),
+            ('pse_0_days',),
+            ('pse_1_7_days',),
+            ('pse_8_14_days',),
+            ('pse_15_20_days',),
+            ('pse_21_days',),
+            ('thr_eligible',),
+            ('thr_0_days',),
+            ('thr_1_7_days',),
+            ('thr_8_14_days',),
+            ('thr_15_20_days',),
+            ('thr_21_days',),
+            ('lunch_eligible',),
+            ('lunch_0_days',),
+            ('lunch_1_7_days',),
+            ('lunch_8_14_days',),
+            ('lunch_15_20_days',),
+            ('lunch_21_days',),
+            ('state_is_test', 'MAX(state_is_test)'),
+            (
+                'district_is_test',
+                lambda col: 'MAX({column})'.format(column=col) if aggregation_level > 1 else "0"
+            ),
+            (
+                'block_is_test',
+                lambda col: 'MAX({column})'.format(column=col) if aggregation_level > 2 else "0"
```

Comment:
nit: is there any reason to make these callables? i find it adds a bit of maybe unneeded complexity. is it just so that you can copy/paste the `MAX({column})` part?

Reply:
Yeah, there is no reason. I realize I could just pull this part as a separate function to remove the code duplication.